analytics.js

```diff
@@ -193,10 +193,22 @@ var Analytics = {
         // data.rows is missing if there are no results
         if (data.totalResults > 0) {
 
+            // get indices of column headers, for reference in client-side thresholds
+            var columnIndices = {};
+            for (var c=0; c<data.columnHeaders.length; c++)
+                columnIndices[data.columnHeaders[c].name] = c;
+
             // Calculate each individual data point.
             for (var i=0; i<data.rows.length; i++) {
                 var row = data.rows[i];
 
+                // allow client-side imposition of a value threshold, where it can't
+                // be done at the server-side (e.g. metric filters on RT reports)
+                if (report.threshold) {
+                    if (parseInt(row[columnIndices[report.threshold.field]]) < report.threshold.value)
+                        continue;
+                }
+
```

Comment:
The Google Analytics Real Time Reporting API doesn't allow for server-side filters on metrics (e.g. limiting results to those with a certain number of active visitors). So, the above code implements a client-side "threshold" feature, where a report can specify a client-side filter like this:

```json
"threshold": {
  "field": "rt:activeUsers",
  "value": "10"
}
```